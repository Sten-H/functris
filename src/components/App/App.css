.App {
    text-align: center;
}

.App-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 80px;
}

.App-header {
    background-color: #222;
    height: 150px;
    padding: 20px;
    color: white;
}

.App-title {
    font-size: 1.5em;
}

.App-intro {
    font-size: large;
}

/*
 FIXME probably shouldn't use margin for this, instead wrap in container and use padding, this way it won't mess
 with total width of columns I think
*/

@keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
}