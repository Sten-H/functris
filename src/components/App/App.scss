.App {
    text-align: center;
}
.game-info .info-text {
    font-weight: 500;
    font-size: 1.2em;
}
/* For some reason these hr are not responding to bootstrap margin classes */
.game-info hr {
    margin-top: 0;
    margin-bottom: 0;
}
.game-info hr.line-top-margin {
    margin-top: 10px;
}

.key {
    border: 2px solid;
}
.key-space {
    content: "Space"
}