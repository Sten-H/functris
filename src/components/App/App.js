import * as React from 'react';
import Bag from '../../containers/Bag/index';
import { Col, Row } from 'react-bootstrap';
import { PauseModal } from '../PauseModal';
import Tetris from '../../containers/Tetris/index';
import GameOverModal from '../../containers/GameOverModal/index';
import { Controls } from '../Controls';
import GameInfo from '../../containers/GameInfo/index';

import './App.css';
// Boxes to the sides of tetris games
const InfoBoxOuter = (props) => (
	<Col className='info-box-wrapper' smOffset={1} sm={3} mdOffset={0} md={4} xsHidden>
		{props.children}
	</Col>
);
export const App = () => {
	return (
		<Row className="App">
			<Col sm={12} xsHidden>
				<h2>TETRIS</h2>
				<hr />
			</Col>
			<InfoBoxOuter >
				<Col sm={12} mdOffset={6} md={6} lgOffset={6} lg={6}>
					<Controls />
				</Col>
			</InfoBoxOuter>
			<Col xs={12} sm={4}>
				<PauseModal />
				<GameOverModal />
				<Tetris />
			</Col>
			<InfoBoxOuter>
				<Col className='game-info solid-border text-left' sm={12} md={6}>
					<Bag />
					<GameInfo />
				</Col>
			</InfoBoxOuter>
		</Row>
	);
};