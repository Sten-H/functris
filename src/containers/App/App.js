import * as React from 'react';
import { connect } from 'react-redux';
import { increment, decrement } from "../../actions/actions"
import Tetris from '../Tetris';
import { Bag } from '../Bag';
import './App.css';
import { Col, ListGroup, ListGroupItem, Row, Table } from "react-bootstrap";

export const App = () => {
    return (
      <div className="App">
          <Row>
              <Col sm={12}>
                  <h2>TETRIS</h2>
                  <hr />
              </Col>
              <Col className="dotted-border hs-info" sm={4} xsHidden>
                  <Col className="float-right"xs={12}>
                      <h4>High Score</h4>
                      <Table className="hs-table" condensed hover>
                          <thead>
                          <tr>
                              <th className="text-muted">#</th>
                              <th className="text-muted">NAME</th>
                              <th className="text-muted">SCORE</th>
                          </tr>
                          </thead>
                          <tbody>
                          <tr>
                              <td>1</td>
                              <td>M4RK</td>
                              <td>120000</td>
                          </tr>
                          <tr>
                              <td>2</td>
                              <td>J4C0B</td>
                              <td>115000</td>
                          </tr>
                          <tr>
                              <td>2</td>
                              <td>J4C0B</td>
                              <td>115000</td>
                          </tr>
                          </tbody>
                      </Table>
                  </Col>
              </Col>
              <Col xs={12} sm={4}>
                  <Tetris />
              </Col>
              <Col className="info-view dotted-border text-left" sm={4} xsHidden>
                  <Bag />
                  <h4>Score</h4>
                  <p>100</p>
                  <h4>Lines</h4>
                  <p>23</p>
              </Col>
          </Row>
      </div>
    );
};
const mapStateToProps = (state) => {
    return {
    };
};

export function mapDispatchToProps(dispatch) {
    return {
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
