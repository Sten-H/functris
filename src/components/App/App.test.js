import * as React from 'react';
import { App, InfoBoxOuter } from './App';
import * as enzyme from 'enzyme';
describe('App component', () => {
    it('Should match snapshot', () => {
        const component = enzyme.shallow(<App />);
        expect(component).toMatchSnapshot();
    });
});
