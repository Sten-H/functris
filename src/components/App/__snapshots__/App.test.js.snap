// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bsClass": "row",
      "children": Array [
        <Col
          bsClass="col"
          componentClass="div"
          sm={12}
          xsHidden={true}
>
          <h2>
                    TETRIS
          </h2>
          <hr />
</Col>,
        <InfoBoxOuter>
          <Col
                    bsClass="col"
                    componentClass="div"
                    lg={6}
                    lgOffset={6}
                    md={6}
                    mdOffset={6}
                    sm={12}
          >
                    <Unknown />
          </Col>
</InfoBoxOuter>,
        <Col
          bsClass="col"
          componentClass="div"
          sm={4}
          xs={12}
>
          <PauseModal />
          <Connect(GameOverModal) />
          <Connect(Tetris) />
</Col>,
        <InfoBoxOuter>
          <Row
                    bsClass="row"
                    componentClass="div"
          >
                    <Col
                              bsClass="col"
                              className="game-info solid-border text-left"
                              componentClass="div"
                              md={6}
                              sm={12}
                    >
                              <Connect(Bag) />
                              <hr
                                        className="line-top-margin"
                              />
                              <h4>
                                        Score
                              </h4>
                              <p
                                        className="info-text"
                              >
                                        100
                              </p>
                              <hr />
                              <h4>
                                        Lines
                              </h4>
                              <p
                                        className="info-text"
                              >
                                        23
                              </p>
                    </Col>
          </Row>
</InfoBoxOuter>,
      ],
      "className": "App",
      "componentClass": "div",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <h2>
              TETRIS
</h2>,
            <hr />,
          ],
          "componentClass": "div",
          "sm": 12,
          "xsHidden": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "TETRIS",
            },
            "ref": null,
            "rendered": "TETRIS",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <Col
            bsClass="col"
            componentClass="div"
            lg={6}
            lgOffset={6}
            md={6}
            mdOffset={6}
            sm={12}
>
            <Unknown />
</Col>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": <Unknown />,
            "componentClass": "div",
            "lg": 6,
            "lgOffset": 6,
            "md": 6,
            "mdOffset": 6,
            "sm": 12,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bsClass": "col",
          "children": Array [
            <PauseModal />,
            <Connect(GameOverModal) />,
            <Connect(Tetris) />,
          ],
          "componentClass": "div",
          "sm": 4,
          "xs": 12,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <Row
            bsClass="row"
            componentClass="div"
>
            <Col
                        bsClass="col"
                        className="game-info solid-border text-left"
                        componentClass="div"
                        md={6}
                        sm={12}
            >
                        <Connect(Bag) />
                        <hr
                                    className="line-top-margin"
                        />
                        <h4>
                                    Score
                        </h4>
                        <p
                                    className="info-text"
                        >
                                    100
                        </p>
                        <hr />
                        <h4>
                                    Lines
                        </h4>
                        <p
                                    className="info-text"
                        >
                                    23
                        </p>
            </Col>
</Row>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <Col
              bsClass="col"
              className="game-info solid-border text-left"
              componentClass="div"
              md={6}
              sm={12}
>
              <Connect(Bag) />
              <hr
                            className="line-top-margin"
              />
              <h4>
                            Score
              </h4>
              <p
                            className="info-text"
              >
                            100
              </p>
              <hr />
              <h4>
                            Lines
              </h4>
              <p
                            className="info-text"
              >
                            23
              </p>
</Col>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": Array [
                <Connect(Bag) />,
                <hr
                  className="line-top-margin"
/>,
                <h4>
                  Score
</h4>,
                <p
                  className="info-text"
>
                  100
</p>,
                <hr />,
                <h4>
                  Lines
</h4>,
                <p
                  className="info-text"
>
                  23
</p>,
              ],
              "className": "game-info solid-border text-left",
              "componentClass": "div",
              "md": 6,
              "sm": 12,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "line-top-margin",
                },
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Score",
                },
                "ref": null,
                "rendered": "Score",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "100",
                  "className": "info-text",
                },
                "ref": null,
                "rendered": "100",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "hr",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Lines",
                },
                "ref": null,
                "rendered": "Lines",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "23",
                  "className": "info-text",
                },
                "ref": null,
                "rendered": "23",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bsClass": "row",
        "children": Array [
          <Col
            bsClass="col"
            componentClass="div"
            sm={12}
            xsHidden={true}
>
            <h2>
                        TETRIS
            </h2>
            <hr />
</Col>,
          <InfoBoxOuter>
            <Col
                        bsClass="col"
                        componentClass="div"
                        lg={6}
                        lgOffset={6}
                        md={6}
                        mdOffset={6}
                        sm={12}
            >
                        <Unknown />
            </Col>
</InfoBoxOuter>,
          <Col
            bsClass="col"
            componentClass="div"
            sm={4}
            xs={12}
>
            <PauseModal />
            <Connect(GameOverModal) />
            <Connect(Tetris) />
</Col>,
          <InfoBoxOuter>
            <Row
                        bsClass="row"
                        componentClass="div"
            >
                        <Col
                                    bsClass="col"
                                    className="game-info solid-border text-left"
                                    componentClass="div"
                                    md={6}
                                    sm={12}
                        >
                                    <Connect(Bag) />
                                    <hr
                                                className="line-top-margin"
                                    />
                                    <h4>
                                                Score
                                    </h4>
                                    <p
                                                className="info-text"
                                    >
                                                100
                                    </p>
                                    <hr />
                                    <h4>
                                                Lines
                                    </h4>
                                    <p
                                                className="info-text"
                                    >
                                                23
                                    </p>
                        </Col>
            </Row>
</InfoBoxOuter>,
        ],
        "className": "App",
        "componentClass": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <h2>
                TETRIS
</h2>,
              <hr />,
            ],
            "componentClass": "div",
            "sm": 12,
            "xsHidden": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "TETRIS",
              },
              "ref": null,
              "rendered": "TETRIS",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <Col
              bsClass="col"
              componentClass="div"
              lg={6}
              lgOffset={6}
              md={6}
              mdOffset={6}
              sm={12}
>
              <Unknown />
</Col>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": <Unknown />,
              "componentClass": "div",
              "lg": 6,
              "lgOffset": 6,
              "md": 6,
              "mdOffset": 6,
              "sm": 12,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bsClass": "col",
            "children": Array [
              <PauseModal />,
              <Connect(GameOverModal) />,
              <Connect(Tetris) />,
            ],
            "componentClass": "div",
            "sm": 4,
            "xs": 12,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <Row
              bsClass="row"
              componentClass="div"
>
              <Col
                            bsClass="col"
                            className="game-info solid-border text-left"
                            componentClass="div"
                            md={6}
                            sm={12}
              >
                            <Connect(Bag) />
                            <hr
                                          className="line-top-margin"
                            />
                            <h4>
                                          Score
                            </h4>
                            <p
                                          className="info-text"
                            >
                                          100
                            </p>
                            <hr />
                            <h4>
                                          Lines
                            </h4>
                            <p
                                          className="info-text"
                            >
                                          23
                            </p>
              </Col>
</Row>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "bsClass": "row",
              "children": <Col
                bsClass="col"
                className="game-info solid-border text-left"
                componentClass="div"
                md={6}
                sm={12}
>
                <Connect(Bag) />
                <hr
                                className="line-top-margin"
                />
                <h4>
                                Score
                </h4>
                <p
                                className="info-text"
                >
                                100
                </p>
                <hr />
                <h4>
                                Lines
                </h4>
                <p
                                className="info-text"
                >
                                23
                </p>
</Col>,
              "componentClass": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": Array [
                  <Connect(Bag) />,
                  <hr
                    className="line-top-margin"
/>,
                  <h4>
                    Score
</h4>,
                  <p
                    className="info-text"
>
                    100
</p>,
                  <hr />,
                  <h4>
                    Lines
</h4>,
                  <p
                    className="info-text"
>
                    23
</p>,
                ],
                "className": "game-info solid-border text-left",
                "componentClass": "div",
                "md": 6,
                "sm": 12,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "className": "line-top-margin",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Score",
                  },
                  "ref": null,
                  "rendered": "Score",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "100",
                    "className": "info-text",
                  },
                  "ref": null,
                  "rendered": "100",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Lines",
                  },
                  "ref": null,
                  "rendered": "Lines",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "23",
                    "className": "info-text",
                  },
                  "ref": null,
                  "rendered": "23",
                  "type": "p",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
